sdk/core/core/src/az_span.c

```diff
@@ -231,15 +231,15 @@ AZ_NODISCARD int32_t az_span_find(az_span source, az_span target)
 
 az_span az_span_copy(az_span destination, az_span source)
 {
-  // Implementations of memmove generally do the right thing when number of bytes to move is 0, even
-  // if the ptr is null, but given the behavior is documented to be undefined, we disallow it as a
-  // precondition.
-  AZ_PRECONDITION_VALID_SPAN(source, 0, false);
-
   int32_t src_size = az_span_size(source);
 
   AZ_PRECONDITION_VALID_SPAN(destination, src_size, false);
 
+  if (src_size == 0 || az_span_ptr(source) == NULL)
```

Comment:
I don't think we need the `az_span_ptr(source) == NULL` check.

`if (src_size == 0)` should be enough to capture both conditions.

Also, please keep/modify the comment and move it right above this line, since it captures why we are even doing the check and not just letting it go through to memmove.

Reply:
fine, I can remove it.
But this goes against what you said before about the more we validate weird cases, the best shape we are XD

Reply:
And what about negative source length? :) Is it a  `precondition(source, 0, true)` then, or the `if` should be `size <= 0` instead of just `== 0`

Reply:
> But this goes against what you said before about the more we validate weird cases, the best shape we are XD

That was about precondition checks (where we are less cost sensitive and being extra cautious to catch dev bugs doesn't hurt), compared to checks in release code which has preconditions turned off, where perf/code size is at a premium. Minimizing and optimizing validation checks especially when trying to make trade offs, makes sense to me.

> And what about negative source length? :) Is it a  `precondition(source, 0, true)` then, or the `if` should be `size <= 0` instead of just `== 0`

I think both of those are good ideas :) (where I prefer the precondition and then `== 0` size check).